src/data/nat/basic.lean

```diff
@@ -940,6 +940,25 @@ strict_mono.lt_iff_lt (pow_right_strict_mono k)
 lemma pow_right_injective {x : ℕ} (k : 2 ≤ x) : function.injective (nat.pow x) :=
 strict_mono.injective (pow_right_strict_mono k)
 
+lemma pow_dvd_pow_iff_pow_le_pow {x k l : ℕ} (w : 0 < x) : x^k ∣ x^l ↔ x^k ≤ x^l :=
+begin
+cases x,
+{ cases w, },
+{ split,
+  { intro a, exact le_of_dvd (pow_pos (succ_pos x) l) a, },
+  { intro a, cases x, simp,
```

Comment:
This nonterminal `simp` isn't good. Also using autogenerated names from `cases` isn't nice in my opinion (even when they are the same as old names; I guess this is debatable). Maybe it would be cleaner to start with the equation compiler?

Reply:
In fact, the simp was terminal, I just hadn't put braces in correctly. This is fixed.

I've switched to using the equation compiler for the outer `cases x`, but not the inner one (which is only needed in one branch of the `iff`).